Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -1264,8 +1264,6 @@ unsigned MediaPlayerPrivateGStreamerBase::videoDecodedByteCount() const
 #if ENABLE(ENCRYPTED_MEDIA)
 void MediaPlayerPrivateGStreamerBase::handleProtectionStructure(const GstStructure* structure)
 {
-    ASSERT(isMainThread());
```

Comment:
This semes to only get called in response to drm-initialization-data-encountered event, I'm surprised we've not tripped it before being called in streaming threads

Reply:
I had the same feeling when I saw it the first time, but here we are answering in the async bus handler, so main thread. Actually, this ASSERT remains in the initialitationDataEncountered method.

Reply:
And the assertion does not hit, I mean.